upgrade/upgrade-instructions-for-improved-data-fixtures-and-tests.md

```diff
@@ -0,0 +1,63 @@
+        ```php
+            foreach ($this->domain as $domainConfig) {
+                $locale = $domainConfig->getLocale();
+                $domainId = $domainConfig->getId();
+                $productData->name[$locale] = t('Name', [], 'dataFixtures', $locale);
+                $productData->description[$domainId] = t('Name', [], 'dataFixtures', $locale);
+            }
+        ```
+    - Remove `demo-data-products.csv` and `demo-data-customers.csv`, along with all the classes responsible for loading data from CSV files and move the data directly to [`ProductDataFixture`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/ProductDataFixture.php) (you need to modify [`Performance/ProductDataFixture`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/src/Shopsys/ShopBundle/DataFixtures/Performance/ProductDataFixture.php) as well) and [`UserDataFixture`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/UserDataFixture.php).
+    - update your [`phpstan.neon`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/phpstan.neon):
+        ```diff
+        -    # A helper methods returning an array of persistent references using $this->getReference()
+        -    message: '#^Method Shopsys\\ShopBundle\\DataFixtures\\ProductDataFixtureReferenceInjector::.+\(\) should return array<.+> but returns array<string, object>\.$#'
+        -    path: %currentWorkingDirectory%/src/Shopsys/ShopBundle/DataFixtures/ProductDataFixtureReferenceInjector.php
+        +    # Actually, we are setting an array item using "$array[] = $this->getReference()"
+        +    message: '#^Array \(array<.+>\) does not accept object\.$#'
+        +    path: %currentWorkingDirectory%/src/Shopsys/ShopBundle/DataFixtures/Demo/ProductDataFixture.php
+        ```
+    - Use [`PriceConverter`](https://github.com/shopsys/shopsys/blob/v8.1.0/packages/framework/src/Model/Pricing/PriceConverter.php) class for converting prices into all currencies, e.g.:
+        ```php
+          foreach ($this->domain->getAllIncludingDomainConfigsWithoutDataCreated() as $domain) {
+              $currency = $this->currencyFacade->getDomainDefaultCurrencyByDomainId($domain->getId());
+              $price = $this->priceConverter->convertPriceWithoutVatToPriceInDomainDefaultCurrency($price, $domain->getId());
+  
+              $transportData->pricesByCurrencyId[$currency->getId()] = $price;
+          }
+        ``` 
+- Modify your functional tests:
+    - Add `skipTestIfFirstDomainIsNotInEnglish`, `getReferenceForDomain`, and `getFirstDomainLocale` methods to your [`FunctionalTestCase`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/tests/ShopBundle/Test/FunctionalTestCase.php).
```

Comment:
You are not saying what is the `getReferenceForDomain` method used for.

Reply:
fixed in 3e53ddd53b6421b5025579dee4a29f0dcefefa85